src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -115,6 +115,93 @@ public static unsafe bool TryGetAsciiString(byte* input, char* output, int count
             return isValid;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        public static unsafe void GetLatin1String(byte* input, char* output, int count)
+        {
+            // Calculate end position
+            var end = input + count;
+
+            do
+            {
+                // If Vector not-accelerated or remaining less than vector size
+                if (!Vector.IsHardwareAccelerated || input > end - Vector<sbyte>.Count)
+                {
+                    if (IntPtr.Size == 8) // Use Intrinsic switch for branch elimination
+                    {
+                        // 64-bit: Loop longs by default
+                        while (input <= end - sizeof(long))
+                        {
+                            output[0] = (char)input[0];
```

Comment:
You could use `Bmi2.X64.ParallelBitDeposit` if supported. Same for the 32-bit path.
See https://github.com/dotnet/aspnetcore/blob/80a77ff5c23647fa52782d15d0abb92cd2afbbb1/src/Shared/ServerInfrastructure/StringUtilities.cs#L121-L137